src/clients/java/src/jni_tests.zig

```diff
@@ -281,28 +281,30 @@ test "JNI: LocalFrame" {
     try testing.expect(boolean_class != null);
     defer env.delete_local_ref(boolean_class);
 
-    var obj = env.alloc_object(boolean_class);
-    defer env.delete_local_ref(obj);
-    try testing.expect(obj != null);
+    const local_ref = env.alloc_object(boolean_class);
+    try testing.expect(local_ref != null);
 
-    // All references must be invalidated after this frame being droped.
-    var pop_local_frame_result = env.pop_local_frame(null);
-    try testing.expect(pop_local_frame_result == null);
+    // All local references must be invalidated after this frame being droped,
+    // except by the frame result.
+    const pop_local_frame_result = env.pop_local_frame(local_ref);
+    try testing.expect(pop_local_frame_result != null);
+    defer env.delete_local_ref(pop_local_frame_result);
 
-    var dead_reference = env.get_object_ref_type(obj);
-    try testing.expect(dead_reference == .invalid);
+    const valid_reference = env.get_object_ref_type(pop_local_frame_result);
+    try testing.expect(valid_reference == .local);
+    try testing.expect(pop_local_frame_result != local_ref);
```

Comment:
The new assertion can't guarantee that the reference was dropped. But it's a JVM detail, our goal is just to assert that the function we called was in fact executed.